tests/eos-test-utils.c

```diff
@@ -0,0 +1,1960 @@
+                              CmdStuff *reaped,
+                              GError **error)
+{
+  g_autofree gchar *bash_script_path = g_strdup (g_getenv ("EOS_CHECK_UPDATER_GDB_BASH_PATH"));
+
+  if (bash_script_path != NULL)
+    return simulated_reap_updater (client,
+                                   cmd,
+                                   reaped,
+                                   error);
+
+  return real_reap_updater (client,
+                            cmd,
+                            reaped,
+                            error);
+}
+
+static gboolean
+run_update_server (GFile *repo,
+                   GFile *quit_file,
+                   guint16 port,
+                   const gchar *remote_name,
+                   CmdAsyncStuff *cmd,
+                   GError **error)
+{
+  SSDEF;
+  g_auto(GStrv) envp = NULL;
+  g_auto(GStrv) argv = NULL;
+  g_autofree gchar *bash_script_path_base = NULL;
+  g_autofree gchar *port_str = g_strdup_printf ("%" G_GUINT16_FORMAT, port);
+
+  envp = generate_strv (SS (envvar ("OSTREE_REPO",
+                                    SS (g_file_get_path (repo)))),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATE_SERVER_QUIT_FILE",
+                                    SS (g_file_get_path (quit_file)))),
+                        NULL);
+
+  argv = generate_strv (EOS_UPDATE_SERVER_BINARY,
+                        SS (flag ("local-port",
+                                  port_str)),
+                        SS (flag ("timeout", "0")),
+                        SS (flag ("serve-remote",
+                                  remote_name)),
+                        NULL);
+
+  bash_script_path_base = g_strdup (g_getenv ("EOS_CHECK_UPDATE_SERVER_GDB_BASH_PATH_BASE"));
+  if (bash_script_path_base != NULL)
+    {
+      g_autoptr(GRegex) regex = g_regex_new ("XXXXXX", 0, 0, error);
+      g_autofree gchar *bash_script_path = NULL;
+      g_autoptr(GFile) bash_script = NULL;
+      g_autofree gchar *delete_me_path = NULL;
+      g_autoptr(GFile) delete_me = NULL;
+
+      if (regex == NULL)
+        return FALSE;
+
+      bash_script_path = g_regex_replace_literal (regex,
+                                                  bash_script_path_base,
+                                                  -1,
+                                                  0,
+                                                  port_str,
+                                                  0,
+                                                  error);
+      if (bash_script_path == NULL)
+        return FALSE;
+
+      bash_script = g_file_new_for_path (bash_script_path);
+      if (!generate_bash_script (bash_script, argv, envp, error))
+        return FALSE;
+
+      delete_me_path = g_strdup_printf ("%s.deleteme", bash_script_path);
+      delete_me = g_file_new_for_path (delete_me_path);
+      g_printerr ("Bash script %s generated, chmod a+x it an run it, make check will continue when %s is deleted\n",
```

Comment:
Is there any way that this can be automated? For instance by using https://developer.gnome.org/gio/stable/GFileInfo.html#G-FILE-ATTRIBUTE-ACCESS-CAN-EXECUTE:CAPS